docs/docs/concepts/memory.md

```diff
@@ -2,23 +2,44 @@
 
 ## What is Memory?
 
-Memory in the context of LLMs and AI applications refers to the ability to process, retain, and utilize information from past interactions or data sources. Examples include:
+Memory in the context of LLMs and AI applications refers to the ability to process, retain, and utilize information from past interactions or data sources. 
+We break this guide into two parts: short term memory and long-term memory.
 
-- Managing what messages (e.g., from a long message history) are sent to a chat model to limit token usage
-- Summarizing past conversations to give a chat model context from prior interactions
-- Selecting few shot examples (e.g., from a dataset) to guide model responses
-- Maintaining persistent data (e.g., user preferences) across multiple chat sessions
-- Allowing an LLM to update its own prompt using past information (e.g., meta-prompting)
-- Retrieving information relevant to a conversation or question from a long-term storage system
+By **short-term memory** we mean memory that persists **within** a single conversation (or interaction) with a user. In LangGraph, we manage short-term memory with the concept of [threads](persistence.md#threads) and [checkpointers](persistence.md#checkpoints)
 
-Below, we'll discuss each of these examples in some detail. 
+By **long-term memory** we mean memory that persists **across** conversations (or interactions) with a user. In LangGraph, we manage long-term memory with [stores](persistence.md#memory-store)
 
-## Managing Messages
+Both are important to understand and implement for your application.
+
+![](img/memory/short-vs-long.png)
+
+## Short-term memory
+
+Short-term memory primarily refers to the ability of an application to accurately remember interactions that previously occurred in the same conversation with the user.
+Most conversations are represented as a list of messages and so the best first thing to do is to store a list of those messages and pass those in to future invocations of the LLM.
+LangGraph's [persistence layer](persistence.md#persistence) has checkpointers that enable [thread](persistence.md#threads)-level memory.
+
+As the conversation grows in length, you will need to think about how to manage that list of messages.
+The reason you will need to think about this is that too many messages will either (a) not fit inside an LLMs context window and will throw an error, or (b) will "overwhelm" the LLM and cause it to perform poorly.
+Therefore, it becomes crucial to think about what parts (or representations) of the conversation to pass to the LLM.
+
+**Note:** we say "messages" and "conversation" above, but short-term memory is not limited to conversations.
```

Comment:
i would make this more prominent with an admonition. that was one of my questions when i was reading the above